### PR TITLE
refactor: factor out DenoSubcommand enum variant into structs

### PR DESCRIPTION
**Pull request by bartlomieju**
Ref https://github.com/denoland/deno/pull/11776#discussion_r699701642

### CODE REVIEW

File: cli/flags.rs [hunk 1]
```diff
@@ -32,97 +32,142 @@ lazy_static::lazy_static! {
...
-    output: Option<PathBuf>,
-    ca_file: Option<String>,
-  },
+  Upgrade(UpgradeFlags),
 }
 
 impl Default for DenoSubcommand {
   fn default() -> DenoSubcommand {
-    DenoSubcommand::Repl { eval: None }
+    DenoSubcommand::Repl(ReplFlags { eval: None })
```
Comment (caspervonb) [inline cli/flags.rs:170]: Maybe we should implement default for the new command flag sets as-well?

File: cli/main.rs [hunk 2]
```diff
@@ -1068,14 +1077,14 @@ async fn test_command(
...
       flags,
-      include,
-      ignore,
-      doc,
-      no_run,
-      fail_fast,
-      filter,
-      shuffle,
-      concurrent_jobs,
+      test_flags.include,
```
Comment (caspervonb) [inline cli/main.rs:1080]: Nice, this is pretty much exactly what I had in mind.
I'll follow up with a cleanup of these signatures.

File: cli/flags.rs [hunk 3]
```diff
@@ -3665,7 +3712,7 @@ mod tests {
     assert_eq!(
       r.unwrap(),
       Flags {
-        subcommand: DenoSubcommand::Test {
+        subcommand: DenoSubcommand::Test(TestFlags {
```
Comment (caspervonb) [inline cli/flags.rs:3715]: Tests could be simpler if we used defaults.

Comment (bartlomieju) [reply cli/flags.rs:3715]: Good point, but I think this could be done in a follow up

Comment (caspervonb) [reply cli/flags.rs:3715]: SGTM.

**Pull request merged**